pkg/authenticate/kubernetes.go

```diff
@@ -125,8 +127,19 @@ func (ki *KubernetesIntermediary) next(ctx context.Context, state *Authenticatio
 		}
 	}
 
+	var stepImage string
+
+	for _, container := range pod.Spec.Containers {
+		if container.Name == model.ActionPodStepContainerName {
```

Comment:
You can actually use `pod.IsContainerStep()` from `github.com/tektoncd/pipeline/pkg/pod` here, might be more future-proof since they've changed the prefix before :/.

Reply:
holding off on this until I have time to resolve a complicated tekton/go-containerregistry version conflict.